### PR TITLE
#256 build updates

### PR DESCRIPTION
**Pull request by rsupnekar**
Removed the nebula.provided-base plugin which is not supported with Gradle 3.4 and above.  Used the suggestion from the nebula plugin developers to use the standard Java compileOnly task.

Also added data binding in test-basic for many views.

### CODE REVIEW

File: sdk/sdk-api/build.gradle [hunk 1]
```diff
@@ -21,7 +20,8 @@ dependencies {
     compileApi project (":cmapi")
     compileApi      ("com.eclipsesource.minimal-json:minimal-json")
     compileApi      ("mil.army.missioncommand:mirrorcache-api")
-    provided        (group: 'mil.army.missioncommand', name: 'mil-sym-android-renderer', ext: 'jar') { transitive = true }
+    compileOnly        (group: 'mil.army.missioncommand', name: 'mil-sym-android-renderer', ext: 'jar') { transitive = true }
```
Comment (stevendlander) [inline sdk/sdk-api/build.gradle]: nit: the group block isnt lined up with the others

File: tests/basic/src/main/java/mil/emp3/dev_test_sdk/MainActivity.java [hunk 2]
```diff
@@ -261,10 +259,9 @@ public void onEvent(MapUserInteractionEvent event) {
                                 pos.getLongitude(), Toast.LENGTH_LONG).show();
                     } catch (EMP_Exception e) {
                         e.printStackTrace();
-                        bFeatureUIEventProcessed = true;
                     }
                     break;
-                case DRAG:
+                default:
                     break;
```
Comment (stevendlander) [inline tests/basic/src/main/java/mil/emp3/dev_test_sdk/MainActivity.java]: Do we have the ability to log this event being essentially dropped or ignored?

File: tests/basic/src/main/java/mil/emp3/dev_test_sdk/MainActivity.java [hunk 3]
```diff
@@ -338,7 +334,7 @@ public void onEvent(FeatureUserInteractionEvent event) {
                         }
                     }
                     break;
-                case DRAG:
+                default:
```
Comment (stevendlander) [inline tests/basic/src/main/java/mil/emp3/dev_test_sdk/MainActivity.java:338]: Same as above comment about logging.

File: tests/basic/src/main/java/mil/emp3/dev_test_sdk/MainActivity.java [hunk 4]
```diff
@@ -538,9 +534,9 @@ protected IGeoPosition getRandomCoordinate() {
         return oPosList;
     }
 
-    private java.util.List<IGeoPosition> getPositions(int iCount, double dAltitude) {
+    private List<IGeoPosition> getPositions(int iCount, double dAltitude) {
```
Comment (stevendlander) [inline tests/basic/src/main/java/mil/emp3/dev_test_sdk/MainActivity.java:539]: Can these be `final` variables?

File: sdk/sdk-api/build.gradle (hunk 1 above)
Comment (rsupnekar) [reply sdk/sdk-api/build.gradle]: fixed

File: tests/basic/src/main/java/mil/emp3/dev_test_sdk/MainActivity.java (hunk 2 above)
Comment (rsupnekar) [reply tests/basic/src/main/java/mil/emp3/dev_test_sdk/MainActivity.java]: added log

File: tests/basic/src/main/java/mil/emp3/dev_test_sdk/MainActivity.java (hunk 3 above)
Comment (rsupnekar) [reply tests/basic/src/main/java/mil/emp3/dev_test_sdk/MainActivity.java:338]: added log

File: tests/basic/src/main/java/mil/emp3/dev_test_sdk/MainActivity.java (hunk 4 above)
Comment (rsupnekar) [reply tests/basic/src/main/java/mil/emp3/dev_test_sdk/MainActivity.java:539]: this is a method, not a variable

Comment (stevendlander) [reply tests/basic/src/main/java/mil/emp3/dev_test_sdk/MainActivity.java:539]: You can't make this:

`private List<IGeoPosition> getPositions(final int iCount, final double dAltitude) {` ?

Comment (stevendlander) [reply tests/basic/src/main/java/mil/emp3/dev_test_sdk/MainActivity.java:539]: In general, I enforce final variables everywhere it is possible.  I know this is outside the scope of this PR, but we should start transitioning that way for the EMP codebase.

File: tests/basic/src/main/java/mil/emp3/dev_test_sdk/MainActivity.java [hunk 5]
```diff
@@ -681,22 +673,22 @@ private void plotManyMilStd(int iMaxCount) {
...
-        java.util.List<IFeature> oFeatureList = new java.util.ArrayList<>();
+        List<IFeature> oFeatureList = new ArrayList<>();
 
         for (String sBasicSymbolCode: oSymbols) {
             //Log.d(TAG, "Symbol " + sBasicSymbolCode);
             if (SymbolUtilities.isWarfighting(sBasicSymbolCode)) {
                 oSymDef = oDefMap.get(sBasicSymbolCode);
 
-                java.util.List<IGeoPosition> oPosList = this.getPositions(1, 20000.0);
+                List<IGeoPosition> oPosList = this.getPositions(1, 20000.0);
```
Comment (stevendlander) [inline tests/basic/src/main/java/mil/emp3/dev_test_sdk/MainActivity.java:693]: Is 20000.0 a magic number?  Should this be a named `const` or something similar?

Comment (rsupnekar) [reply tests/basic/src/main/java/mil/emp3/dev_test_sdk/MainActivity.java:693]: it is a test program, and I didn't change someone else's test

File: tests/basic/src/main/java/mil/emp3/dev_test_sdk/MainActivity.java (hunk 4 above)
Comment (rsupnekar) [reply tests/basic/src/main/java/mil/emp3/dev_test_sdk/MainActivity.java:539]: This program is meant for quick tests, it is not part of the release

File: tests/basic/src/main/java/mil/emp3/dev_test_sdk/MainActivity.java [hunk 6]
```diff
@@ -907,78 +902,62 @@ protected void onCreate(Bundle savedInstanceState) {
...
+    public void editorComplete(View view) {
+        try {
+            switch (MainActivity.this.ePlotMode) {
+                case EDIT_FEATURE:
+                    MainActivity.this.map.completeEdit();
+                    break;
+                case DRAW_FEATURE:
+                    MainActivity.this.map.completeDraw();
+                    break;
+                default:
```
Comment (stevendlander) [inline tests/basic/src/main/java/mil/emp3/dev_test_sdk/MainActivity.java:931]: similar comment regarding logging here

File: tests/basic/src/main/java/mil/emp3/dev_test_sdk/MainActivity.java [hunk 7]
```diff
@@ -1076,336 +1055,247 @@ public void run() {
...
-        // Pan left turns the camera left 5 degrees
-        // each time the button is pressed
-        ImageButton panLeft = (ImageButton) findViewById(R.id.PanLeft);
-        if (panLeft != null) {
-            panLeft.setOnClickListener(v -> {
-                ICamera camera = MainActivity.this.oCamera;
-                try {
-                    double dPan = camera.getHeading();
+        boolean bBatch = MainActivity.this.oBatchUpdateCkb.isChecked();
+        boolean bAff = MainActivity.this.oAffiliationCkb.isChecked();
```
Comment (stevendlander) [inline tests/basic/src/main/java/mil/emp3/dev_test_sdk/MainActivity.java:1122]: Should these be `MainActivity` or `mainBinding`?

**Review by stevendlander — APPROVED** [review]

File: tests/basic/src/main/java/mil/emp3/dev_test_sdk/MainActivity.java (hunk 4 above)
Comment (stevendlander) [reply tests/basic/src/main/java/mil/emp3/dev_test_sdk/MainActivity.java:539]: OK understood.

File: tests/basic/src/main/java/mil/emp3/dev_test_sdk/MainActivity.java (hunk 7 above)
Comment (rsupnekar) [reply tests/basic/src/main/java/mil/emp3/dev_test_sdk/MainActivity.java:1122]: That's correct.  The code hasn't completely covered to data binding yet.

File: tests/basic/src/main/java/mil/emp3/dev_test_sdk/MainActivity.java (hunk 6 above)
Comment (rsupnekar) [reply tests/basic/src/main/java/mil/emp3/dev_test_sdk/MainActivity.java:931]: added

**Pull request merged**